src/main/java/com/midaswebserver/midasweb/services/UserServiceImp.java

```diff
@@ -0,0 +1,264 @@
+        if (username == null) {
+            log.error("getIDByUser: passed username was null");
+            return null;
+        }
+        if (users.size() == 0) {
+            log.info("getIDByUser: no user found with username of '{}'", username);
+            return null;
+        }
+        if (users.size() > 1) {
+            log.error("getIDByUser: two users with username '{}' have been found", username);
+            return null;
+        }
+        return users.get(0).getId();
+    }
+
+    /**
+     * Updates the user in database, make sure that the param has the original user ID set as its ID,
+     * the function will return null if not
+     *
+     * @param user
+     * @return boolean
+     */
+    @Override
+    public boolean update(User user) {
+        if (user == null) {
+            log.error("update: passed username was null");
+            return false;
+        }
+        if (this.getUserById(user.getId()) == null) {
+            log.warn("update: D of user '{}' didn't match anything in database", user.getUsername());
+            return false;
+        }
+        log.debug("update: User '{}' has been updated", user.getUsername());
+        userRepo.save(user);
+        return true;
+    }
+
+    /**
+     * Adds a symbol to a user. This will create a symbol object linked to the user object
+     * which can be used to retrieve the saved ticker
+     * WIll return false if user or ticker is null
+     * if ticker is invalid > 4, return false
+     * if User doesn't exist, return false
+     * if ticker is already saved, return true
+     * if ticker is added, return true
+     * all else fails, return false
+     *
+     * @param user
+     * @param ticker like 'UEC'
+     * @return boolean based on the success of the transaction
+     */
+    @Transactional
+    @Override
+    public boolean addSymbolToUser(User user, String ticker) {
+        //validation
+        if (user == null || ticker == null) {
+            log.error("addSymbolToUser: user, '{}', or symbol, '{}', was found to be null", user, ticker);
+            return false;
+        }
+        if (ticker.getBytes().length > 4) {
+            log.error("addSymbolToUser: symbol, '{}', was longer than expected, 4 characters", ticker);
+            return false;
+        }
+        if (!userRepo.existsById(user.getId())) {
+            log.error("addSymbolToUser: user, '{}', with id, '{}', was not found in database", user.getUsername(), user.getId());
+            return false;
+        }
+        //logic
+        for (Symbol symbolSaved : user.getSymbol()) {
+            if (symbolSaved.getTicker().equalsIgnoreCase(ticker)) {
+                log.debug("addSymbolToUser: user, '{}', attempted to add ticker, '{}', that was added previously", user.getUsername(), ticker);
+                return true;
+            }
+        }
+        user.addTicker(ticker, user);
+        update(user);
+        for (Symbol symbolSaved : user.getSymbol()) {
```

Comment:
This feels more like something you would do in a unit test, not in the actual service.